src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1138,9 +1138,19 @@ private string GetLogicForDefaultSerializerOptionsInit()
             {
                 JsonSourceGenerationOptionsAttribute options = _currentContext.GenerationOptions;
 
-                string? namingPolicyInit = options.PropertyNamingPolicy == JsonKnownNamingPolicy.CamelCase
+                string? namingPolicyName = options.PropertyNamingPolicy switch
+                {
+                    JsonKnownNamingPolicy.CamelCase => "CamelCase",
+                    JsonKnownNamingPolicy.SnakeLowerCase => "SnakeLowerCase",
+                    JsonKnownNamingPolicy.SnakeUpperCase => "SnakeUpperCase",
+                    JsonKnownNamingPolicy.KebabLowerCase => "KebabLowerCase",
+                    JsonKnownNamingPolicy.KebabUpperCase => "KebabUpperCase",
+                    _ => null,
+                };
+
+                string? namingPolicyInit = namingPolicyName != null
```

Comment:
How about using the null-coaleshing operator (`??`)?

Reply:
I don't think that will work here, he uses interpolation when it's not null